src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -96,5 +86,263 @@ public bool NeedsRefresh()
+                    }
+
+                    string propertyName = reader.GetString();
+
+                    switch (propertyName)
+                    {
+                        case FeatureManagementConstants.IdJsonPropertyName:
+                            {
+                                if (reader.Read() && reader.TokenType == JsonTokenType.String)
+                                {
+                                    featureFlag.Id = reader.GetString();
+                                }
+                                else if (reader.TokenType != JsonTokenType.Null)
+                                {
+                                    throw CreateFeatureFlagFormatException(
+                                        FeatureManagementConstants.IdJsonPropertyName,
+                                        settingKey,
+                                        reader.TokenType.ToString(),
+                                        JsonTokenType.String.ToString());
+                                }
+
+                                break;
+                            }
+
+                        case FeatureManagementConstants.EnabledJsonPropertyName:
+                            {
+                                if (reader.Read() && (reader.TokenType == JsonTokenType.False || reader.TokenType == JsonTokenType.True))
+                                {
+                                    featureFlag.Enabled = reader.GetBoolean();
+                                }
+                                else if (reader.TokenType == JsonTokenType.String && bool.TryParse(reader.GetString(), out bool enabled))
+                                {
+                                    featureFlag.Enabled = enabled;
+                                }
+                                else
+                                {
+                                    throw CreateFeatureFlagFormatException(
+                                        FeatureManagementConstants.EnabledJsonPropertyName,
+                                        settingKey,
+                                        reader.TokenType.ToString(),
+                                        $"{JsonTokenType.True}' or '{JsonTokenType.False}");
+                                }
+
+                                break;
+                            }
+
+                        case FeatureManagementConstants.ConditionsJsonPropertyName:
+                            {
+                                if (reader.Read() && reader.TokenType == JsonTokenType.StartObject)
+                                {
+                                    featureFlag.Conditions = ParseFeatureConditions(ref reader, settingKey);
+                                }
+                                else if (reader.TokenType == JsonTokenType.Null)
+                                {
+                                    break;
+                                }
+                                else
```

Comment:
Can we use the same pattern as above?

```
 else if (reader.TokenType != JsonTokenType.Null)
{
    CreateFeatureFlagFormatException
}
```

Reply:
Oh I tried to go back and use the same pattern everywhere but I missed this, I'll do that